src/http.ts

```diff
@@ -0,0 +1,134 @@
+import fetch from 'unfetch';
+
+import {
+  DEFAULT_FETCH_TIMEOUT_MS,
+  DEFAULT_SILENT_TOKEN_RETRY_COUNT
+} from './constants';
+
+import { sendMessage } from './worker/worker.utils';
+import { FetchOptions } from './global';
+
+export const createAbortController = () => new AbortController();
+
+const switchFetch = async (
+  url: string,
+  audience: string,
+  scope: string,
+  opts: FetchOptions,
+  timeout: number,
+  worker?: Worker
+) => {
```

Comment:
We could end up making this one generic, to have a typed response.

Reply:
Sure - maybe one to consider if/when we start calling more than just the token endpoint?

Reply:
Yep, which leaves the question: should this have a  (non geberic) return type?

Reply:
TypeScript infers it as `Promise<unknown>`, which is fair considering we don't know what we'll get back from the server. I'll type it as that for now.

Reply:
I've typed it as `Promise<any>` and made `oauthToken()` generic, gives us the opportunity to pass a type in.

Reply:
I dont think I see oauthToken being generic? I also do not think it should be tho.

Reply:
Yes, I have since updated the code but not replied back here. I've now made `getJSON` generic, and `oauthToken` specifies a new `TokenEndpointResponse` type.

Reply:
Or at least I should have... I'm missing a commit somewhere 🤔

Reply:
I've updated this now.